src/TiledArray/dist_eval/dist_eval.h

```diff
@@ -118,9 +118,10 @@ class DistEvalImpl : public TensorImpl<Policy>,
     set_counter_ = 0;
 
     if (perm) {
-      Permutation inv_perm(-perm);
+      auto outer_perm = perm.outer_permutation();
+      Permutation inv_perm(-outer_perm);
       range_type source_range = inv_perm * trange.tiles_range();
-      source_to_target_ = PermIndex(source_range, perm);
+      source_to_target_ = PermIndex(source_range, outer_perm);
       target_to_source_ = PermIndex(trange.tiles_range(), inv_perm);
```

Comment:
@efv here's one example of a place where we no longer have access to the `VariableList`, but need to separate the permutations.

Reply:
I understand. Using strong type (`BipartitePermutation`) distinct from `Permutation` is still a good idea since the latter is used in many contexts the former does not make sense (e.g. permute `Range` etc.)

I'm working on this.